test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -379,41 +381,22 @@ func testSecondaryNetwork(t *testing.T, networkType string, pods []*testPodInfo)
 
 	testData := &testData{e2eTestData: e2eTestData, networkType: networkType, pods: pods}
 
-	t.Run("testCreateTestPodOnNode", func(t *testing.T) {
-		err := testData.createPods(t, e2eTestData.GetTestNamespace())
-		require.NoError(t, err, "Error when create test Pods")
-	})
-	t.Run("testPingBetweenInterfaces", func(t *testing.T) {
-		err := testData.pingBetweenInterfaces(t)
-		require.NoError(t, err, "Error when pinging between interfaces")
-	})
-	t.Run("testReconcilationAfterAgentRestart", func(t *testing.T) {
-		testData.reconcilationAfterAgentRestart(t)
-	})
-}
-
-func TestSriovNetwork(t *testing.T) {
-	// Create Pods on the control plane Node, assuming a single Node cluster for the SR-IOV
-	// test.
-	nodeName := antreae2e.NodeName(0)
-	pods := []*testPodInfo{
-		{
-			podName:           "sriov-pod1",
-			nodeName:          nodeName,
-			interfaceNetworks: map[string]string{"eth1": "sriov-net1", "eth2": "sriov-net2"},
-		},
-		{
-			podName:           "sriov-pod2",
-			nodeName:          nodeName,
-			interfaceNetworks: map[string]string{"eth2": "sriov-net1", "eth3": "sriov-net3"},
-		},
-		{
-			podName:           "sriov-pod3",
-			nodeName:          nodeName,
-			interfaceNetworks: map[string]string{"eth4": "sriov-net1"},
-		},
+	if err := testData.createPods(t, e2eTestData.GetTestNamespace()); err != nil {
+		t.Fatalf("Error when create test Pods: %v", err)
+	}
+	clientset, err := kubernetes.NewForConfig(e2eTestData.KubeConfig)
+	if err != nil {
+		t.Fatalf("Error when creating kubernetes client: %v", err)
+	}
+	if err := testData.assertPodAnnotation(t, pods, clientset); err != nil {
+		t.Fatalf("Error when checking the Pod annotation: %v", err)
+	}
+	if err := testData.pingBetweenInterfaces(t); err != nil {
+		t.Fatalf("Error when pinging between interfaces: %v", err)
+	}
+	if err := testData.reconcilationAfterAgentRestart(t); err != nil {
+		t.Fatalf("Error when testReconcilationAfterAgentRestart: %v", err)
```

Comment:
The message would be confusing since you removed the subtest

Reply:
Updated.